File: Chapter-2/hello.go
// go run hello.go

package main

import (
  "fmt"
  "bufio"
  "os"
  "strings"
)

func main() {
  fmt.Print("What is your name? ")
  reader := bufio.NewReader(os.Stdin)
  name, _ := reader.ReadString('\n')
  fmt.Printf("Hello, %s, nice to meet you!\n", strings.TrimSpace(name))
}

File: Chapter-2/1/hello.c
#include <stdio.h>

int main() {
  char name[256];
  printf("What is your name? ");
  if (scanf("%s", &name[0]) != 1) {
    return -1;
  }
  printf("Hello, %s, nice to meet you!\n", name);
  return 0;
}
